numpy/core/_exceptions.py

```diff
@@ -106,13 +117,24 @@ def __init__(self, ufunc, casting, from_, to, i):
         self.out_i = i
 
     def __str__(self):
-        # only show the number if more than one output exists
-        i_str = "{} ".format(self.out_i) if self.ufunc.nout != 1 else ""
+        if self.ufunc.nout > 1:
+            from_outs = "out=({})".format(", ".join(f.name for f in self.to[self.ufunc.nin:]))
+            to_outs = "out=({})".format(", ".join(t.name for t in self.to[self.ufunc.nin:]))
+        else:
+            from_outs = "out={}".format(self.from_[-1])
+            to_outs = "out={}".format(self.to[-1])
+
         return (
```

Comment:
Might be worth a helper function somewhere,
```python
def _ufunc_call_sig(ufunc, arg_dtypes):
    if ufunc.nout > 1:
        outs = "out=({})".format(", ".join(f.name for f in arg_dtypes[ufunc.nin:]))
    else:
        outs = "out={}".format(arg_dtypes[-1])
    return f"{}({}, {})".format(ufunc.__name__, ", ".join(arg_dtypes[:ufunc.nin]), outs)
```
which you can then call as `_ufunc_call_sig(self.ufunc, self.from)` and `_ufunc_call_sig(self.ufunc, self.to)` in the error messages

Reply:
I do like this concept, I'm putting it together now to see how that works out. Helper function likely good, but either way something to have the meanings of from/to not switching around is a good thing.

Reply:
> I'm putting it together now to see how that works out.

Were you able to finish this up?